compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -344,14 +344,14 @@ impl<'cx, 'tcx> FallibleTypeFolder<'tcx> for QueryNormalizer<'cx, 'tcx> {
                 // `tcx.normalize_projection_ty` may normalize to a type that still has
                 // unevaluated consts, so keep normalizing here if that's the case.
                 if res != ty && res.has_type_flags(ty::TypeFlags::HAS_CT_PROJECTION) {
-                    Ok(res.try_super_fold_with(self)?)
+                    res.try_super_fold_with(self)?
                 } else {
-                    Ok(res)
+                    res
                 }
             }
 
-            _ => ty.try_super_fold_with(self),
-        })()?;
```

Comment:
Very weird, I hope there wasn't something weird going on, but the change looks correct🤨. Well, there are quite a few of those around, I guess it's just an artifact of refactorings and made sense at some point.

Reply:
Yeah, when I was removing these I was triple checking in case I've missed something ><